app/services/bulk_invoice_service.rb

```diff
@@ -0,0 +1,56 @@
+class BulkInvoiceService
+  include WickedPdf::PdfHelper
+  attr_reader :id
+
+  def initialize
+    @id = new_invoice_id
+  end
+
+  def create_bulk_invoice(order_ids)
+    pdf = CombinePDF.new
+    orders = Spree::Order.where(id: order_ids)
+
+    orders.each do |order|
+      invoice = renderer.render_to_string pdf: "invoice-#{order.number}.pdf",
+                                          template: invoice_template,
+                                          formats: [:html], encoding: "UTF-8",
+                                          locals: { :@order => order }
+
+      pdf << CombinePDF.parse(invoice)
+    end
+
+    pdf.save "#{file_directory}/#{@id}.pdf"
+  end
+  handle_asynchronously :create_bulk_invoice
```

Comment:
:see_no_evil: I find this too implicit and hidden. No one can tell from the consumer (the controller in this case) that this is executed in background and I think it's relevant, especially when debugging (and with background jobs, that happens often :trollface: )

I'd rather use a custom job class and call it as `Delayed::Job.enqueue(BulkInvoiceJob)`. Check the docs: https://github.com/collectiveidea/delayed_job/#custom-jobs

Reply:
I had a custom job at the beginning, but when I was refactoring it seemed really unnecessary, so I cut it. I can move it back to a custom job if you like, or maybe I could rename the method so it's clear it's a background job? `:start_pdf_job` or something similar?